Lib/pathlib.py

```diff
@@ -438,24 +434,23 @@ def __init__(self, pat, child_parts, flavour):
     def _select_from(self, parent_path, is_dir, exists, scandir):
         try:
             with scandir(parent_path) as scandir_it:
-                entries = list(scandir_it)
```

Comment:
i believe this was done to be able to actually end the context manager asap,
please crosscheck the effect, i suspect this may collect excess file descriptors in certain cases

Reply:
Previously it would create a list of all the entries and would use high memory and now there is no overhead as it is just iterating over it, since it is a context manager it will be closed even if an exception occurs.

Reply:
It also means that the results are computed before each file is yielded, so if the caller modifies the filesystem in between you could get different results. Though the context manager is going to close it at some point, how you have it now the `yield` is inside the context manager, meaning that it'll remain open while the caller is using the values.

Reply:
While I reverted the change, it isn't as efficient as it should be.